user-interface/src/lib/components/combobox/ComboBox.tsx

```diff
@@ -0,0 +1,450 @@
+import '../uswds/forms.scss';
+import './ComboBox.scss';
+import {
+  forwardRef,
+  PropsWithChildren,
+  ReactElement,
+  useEffect,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from 'react';
+import Icon from '../uswds/Icon';
+import Button, { UswdsButtonStyle } from '../uswds/Button';
+import PillBox from '../PillBox';
+import useOutsideClick from '@/lib/hooks/UseOutsideClick';
+import { ComboBoxRef } from '@/lib/type-declarations/input-fields';
+
+export type ComboOption = {
+  value: string;
+  label: string;
+  selected?: boolean;
+  hidden?: boolean;
+};
+
+type InputProps = JSX.IntrinsicElements['input'] &
+  JSX.IntrinsicElements['select'] &
+  PropsWithChildren;
+
+export interface ComboBoxProps extends PropsWithChildren, Omit<InputProps, 'onChange'> {
```

Comment:
`PropsWithChildren` is redundant here because it is already included on `InputProps`.